test/config.test.ts

```diff
@@ -0,0 +1,21 @@
+import {evm_revert} from './helper';
+import {getInitialSetup, IInitialSetup} from './setup';
+import {ethers} from 'hardhat';
+
+describe('config test', async () => {
+  let setup: IInitialSetup;
+
+  before(async () => {
+    setup = await getInitialSetup();
+  });
+
+  beforeEach(async () => {
+    await evm_revert(setup.postSetupSnapshotId);
```

Comment:
snapshotId actually can only be used once, subsequence calls with the same id has no effect, the returned result is actually false.

If the desired effect is to revert evm state before each call, we must call evm_snapshot after evm_revert:
```
...
setup.postSetupSnapshotId = await evm_snapshot();
```

Reply:
this line is not so important, commented out